src/prpy/util.py

```diff
@@ -914,6 +914,204 @@ def ComputeUnitTiming(robot, traj, env=None):
+    """
+    Check if a configuration is within a robot's joint position limits.
+
+    If outside limits, this procedure throws an exception
+    of type JointLimitError.
+
+    @param openravepy.robot robot: The robot.
+    @param list             q:     List or array of joint positions.
+    """
+    from prpy.planning.exceptions import JointLimitError
+
+    q_limit_min, q_limit_max = robot.GetActiveDOFLimits()
+    active_dof_indices = robot.GetActiveDOFIndices()
+
+    if len(q) != len(active_dof_indices):
+        raise ValueError('The number of joints in the configuration q '
+                         'is not equal to the number of active DOF.')
+
+    lower_position_violations = (q < q_limit_min)
+    if lower_position_violations.any():
+        index = lower_position_violations.nonzero()[0][0]
+        raise JointLimitError(robot,
+            dof_index=active_dof_indices[index],
+            dof_value=q[index],
+            dof_limit=q_limit_min[index],
+            description='position')
+
+    upper_position_violations = (q > q_limit_max)
+    if upper_position_violations.any():
+        index = upper_position_violations.nonzero()[0][0]
+        raise JointLimitError(robot,
+            dof_index=active_dof_indices[index],
+            dof_value=q[index],
+            dof_limit=q_limit_max[index],
+            description='position')
+
+
+def ConvertIntToBinaryString(x, reverse=False):
+    """
+    Convert an integer to a binary string.
+
+    Optionally reverse the output string, which is
+    required for producing a Van Der Corput sequence.
+
+    @param int  x:       The number to be converted.
+    @param bool reverse: If True, the output string will be reversed.
+
+    @returns string: A binary number as a string.
+    """
+    if type(x) != int:
+        raise ValueError('Input number must be an integer')
+
+    if reverse == True:
+        return ''.join(reversed(bin(x)[2:]))
+
+    return ''.join(bin(x)[2:])
+
+
+def VanDerCorputSequence(lower=0.0, upper=1.0, include_endpoints=True):
+    """
+    Generate the binary Van der Corput sequence, where each value
+    is a dyadic fraction re-scaled to the desired range.
+
+    For example, on the interval [0,1], the first 5 values of
+    the Van der Corput sequence are:
+    [0.0, 1.0, 0.5, 0.5, 0.75]
+
+    @param float lower: The first value of the range of the sequence.
+    @param float upper: The last value of the range of the sequence.
+
+    @param bool include_endpoints: If True, the output sequence will
+                                   include the value 'lower' and the
+                                   value 'upper'.
+                                   If False, these endpoint values
+                                   will not be returned.
+
+    @returns generator: A sequence of float values.
+    """
+    from itertools import count, chain
+
+    if include_endpoints == True:
+        endpoints = (0.0, 1.0)
+    else:
+        endpoints = None
```

Comment:
If you set `endpoints = []` here you can skip the `endpoints or []` check below.